compiler/lexer.nim

```diff
@@ -74,11 +75,15 @@ type
 
   TokTypes* = set[TokType]
 
+when defined(nimsuggest):
```

Comment:
why `when defined(nimsuggest):` ? (previous code didn't have it)

Reply:
It suppresses a warning message from the compiler. `weakTokens` is only used when `nimsuggest` is defined.

Admittedly, it is off-topic from the commit, but it removes one more warning to see.

Reply:
ok!